### PR TITLE
Recipe app development

### PR DESCRIPTION
**Pull request by rolandfbk**
Recipe app project is implemented following the Microverse requirements below:

Project requirements

- Login page and registration page:

Should be built with Devise.

- Food list:

Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
Should lead to a form that allows users to add new food.

- Recipes list:

Should display a list of recipes created by the logged-in user as in the wireframe.
Should lead to recipe details.
If the user is the owner of the recipe, should allow the user to delete it.

- Public recipe list:

Should display a list of all public recipes ordered by newest as in the wireframe.
Should lead to recipe details.

- Recipe details:

If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

- General shopping list view (only for a group that has less than 3 members):

Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
Should count the total food items and total price of the missing food.

- Make sure there are no N+1 queries happening.

- Create a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: app/controllers/general_shopping_list_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @recipe = Recipe.find(params[:recipe])
+    @recipe_foods = @recipe.recipe_foods.includes(:food)
+    @food_items_to_buy = []
+    @recipe_foods.each do |rfood|
+      item = {
+        name: rfood.food.name,
+        quantity: rfood.food.quantity < rfood.quantity ? rfood.quantity - rfood.food.quantity : rfood.quantity,
+        price: rfood.food.price,
+        measurement_unit: rfood.food.measurement_unit
+      }
```
Comment (AnselemOdims) [inline app/controllers/general_shopping_list_controller.rb:17]: - [ ] At the moment, the general shopping list contains food added to the recipe but it should only show the list of food that is missing for all recipes of the logged-in user. Please kindly adjust to match the requirements.
- [ ] It should also count the total food items and total price of the missing food.

File: app/controllers/general_shopping_list_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def index
+    @recipe = Recipe.find(params[:recipe])
+    @recipe_foods = @recipe.recipe_foods.includes(:food)
+    @food_items_to_buy = []
+    @recipe_foods.each do |rfood|
+      item = {
+        name: rfood.food.name,
+        quantity: rfood.food.quantity < rfood.quantity ? rfood.quantity - rfood.food.quantity : rfood.quantity,
+        price: rfood.food.price,
```
Comment (AnselemOdims) [inline app/controllers/general_shopping_list_controller.rb:15]: - [ ] Please kindly make sure most of the business logic are put in the models thereby making the controllers as thin as possible

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @team, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by elisha2kyakpo1 — COMMENTED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You have done a great job working on the requested changes ma✔️✔️

## Optional suggestions

- [ ] Let's be more careful next time to avoid merging before approval, and it's good that you communicated.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**